### ISSUE TITLE
t/54.glob.t fails on perl 5.8.8 or lower

### ISSUE THREAD

**Issue by skaji** [perlpunk/YAML-PP-p5#39]
perl 5.8.8 or lower has a bug related `qr//m` (see https://github.com/Perl/perl5/issues/1783).
That is, on perl 5.8.8 or lower, the following script prints "not match"
```perl
#!/usr/bin/env perl
use strict;
use warnings;

my $str = <<'EOF';
a
b
EOF

print $str =~ qr/a$/m ? "match" : "not match", "\n";
```

As a result, `t/54.glob.t` fails on perl 5.8.8 or lower.
```
❯ perl -v
This is perl, v5.8.8 built for darwin-2level

❯ prove -l t/54.glob.t
t/54.glob.t .. 1/?
    #   Failed test 'IO::Scalar fileno correct'
    #   at t/54.glob.t line 181.
    #                   '--- !perl/glob:IO::File
    # IO:
    #   fileno: 4
    #   stat:
    #     atime: 1607237357
    #     blksize: 4096
    #     blocks: 16
    #     ctime: 1607237355
    #     device: 16777221
    #     gid: 20
    #     inode: 26870573
    #     links: 1
    #     mode: 33188
    #     mtime: 1607237355
    #     rdev: 0
    #     size: 5126
    #     uid: 501
    #   tell: 0
    # NAME: GEN0
    # PACKAGE: Symbol
    # '
    #     doesn't match '(?m-xis:fileno: 4$)'
    # Looks like you failed 1 test of 2.

#   Failed test 'ioscalar'
#   at t/54.glob.t line 182.
# Looks like you failed 1 test of 3.
t/54.glob.t .. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests

Test Summary Report
-------------------
t/54.glob.t (Wstat: 256 Tests: 3 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
Files=1, Tests=3,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.09 cusr  0.01 csys =  0.13 CPU)
Result: FAIL
```

**Comment by perlpunk** [comment]
Sorry for the late answer, I have been extremely busy.

This is now fixed in master, tested on 5.8.8.
I simply replaced the `$` in the regex with `\n`.

**Comment by skaji** [comment]
@perlpunk Thanks!
